src/GraphQL/Types/Composite/AutoRegisteringHelper.cs

```diff
@@ -303,5 +303,22 @@ internal static TypeInformation GetTypeInformation(ParameterInfo parameterInfo)
             typeInformation.ApplyAttributes();
             return typeInformation;
         }
+
+        /// <summary>
+        /// Identifies the constructor to use when constructing instances of <typeparamref name="TSourceType"/>.
+        /// Selects any public constructor marked with <see cref="GraphQLConstructorAttribute"/>, or the public
+        /// parameterless constructor, or the only public contructor, or returns <see langword="null"/> otherwise.
+        /// </summary>
+        internal static ConstructorInfo? GetConstructor<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TSourceType>()
+        {
+            var constructors = typeof(TSourceType).GetConstructors();
+            return constructors.Length switch
+            {
+                0 => null,
+                1 => constructors[0],
+                _ => Array.Find(constructors, c => c.GetCustomAttribute<GraphQLConstructorAttribute>() != null)
```

Comment:
`Array.Find` is equivalent to `FirstOrDefault`. I think it should throw if multiple constructors are annotated with `GraphQLConstructorAttribute`. The same in the `ToObject` extension method. The analyzer already detects it as an error.

Reply:
In this situation I don’t want it to throw because if someone has overridden ParseDictionary then it should not throw. Note that this is an internal method. 

Yes I wrote the code with FirstOrDefault but the .NET Analyzer suggested changing it to Array.Find so I accepted its suggestion.

Reply:
I could rename the method “OrDefault”?

Reply:
Oh you said if multiple constructors are marked. Yes I can change that.

Reply:
> I don’t want it to throw because if someone has overridden ParseDictionary

But I thought this feature is to avoid the manual definition of `InputObjectGraphType`'s, so there is no place to override `ParseDictionary`

Reply:
Overriding `ParseDictionary` is still possible.  It is certainly viable to derive from `AutoRegisteringInputObjectGraphType<MyInputClass>` and only override `ParseDictionary` if that's the only change that the developer wants to make.  They would not need to define all the fields.